### PR TITLE
add shortcuts

### PR DESCRIPTION
**Pull request by artemanufrij**
`F5` build & run; `ctrl+s` save current document; `ctrl+shift+s` save all documents

### CODE REVIEW

File: src/IDEApplication.vala [hunk 1]
```diff
@@ -45,6 +45,31 @@ public class IDEApplication : Granite.Application {
...
+
+        var save_action = new SimpleAction ("save", null);
+        add_action (save_action);
+        add_accelerator ("<Control>s", "app.save", null);
+        save_action.activate.connect (() => {
+            if (get_main_window () != null) {
+                get_main_window ().save_current_document ();
+            }
+        });
+        var save_all_action = new SimpleAction ("save_all", null);
```
Comment (donadigo) [inline src/IDEApplication.vala]: The ID should be `save-all`.

File: src/IDEApplication.vala [hunk 2]
```diff
@@ -45,6 +45,31 @@ public class IDEApplication : Granite.Application {
         about_license_type = Gtk.License.GPL_3_0;
 
         flags |= GLib.ApplicationFlags.HANDLES_OPEN;
+
+        var save_action = new SimpleAction ("save", null);
+        add_action (save_action);
+        add_accelerator ("<Control>s", "app.save", null);
+        save_action.activate.connect (() => {
+            if (get_main_window () != null) {
```
Comment (donadigo) [inline src/IDEApplication.vala]: Cache the main window like this:
```
var window = get_main_window ();
if (window != null) {
    window.save_current_document ()
}
```

File: src/IDEApplication.vala [hunk 3]
```diff
@@ -45,6 +45,31 @@ public class IDEApplication : Granite.Application {
...
+                get_main_window ().save_current_document ();
+            }
+        });
+        var save_all_action = new SimpleAction ("save_all", null);
+        add_action (save_all_action);
+        add_accelerator ("<Control><Shift>s", "app.save_all", null);
+        save_all_action.activate.connect (() => {
+            if (get_main_window () != null) {
```
Comment (donadigo) [inline src/IDEApplication.vala]: Same here.

File: src/IDEApplication.vala [hunk 4]
```diff
@@ -45,6 +45,31 @@ public class IDEApplication : Granite.Application {
...
+                get_main_window ().save_opened_documents ();
+            }
+        });
+        var build_and_run_action = new SimpleAction ("build_and_run", null);
```
Comment (donadigo) [inline src/IDEApplication.vala]: This ID should be `build-and-run`.

File: src/IDEApplication.vala [hunk 5]
```diff
@@ -45,6 +45,31 @@ public class IDEApplication : Granite.Application {
...
+        add_action (build_and_run_action);
+        add_accelerator ("F5", "app.build_and_run", null);
+        build_and_run_action.activate.connect (() => {
```
Comment (donadigo) [inline src/IDEApplication.vala]: Same here.

**Review by donadigo — CHANGES_REQUESTED** [review]
Some diff comments. In addition to that, remove the `.elementary-ide.proj` and `src/config.vala` files.

**Comment by artemanufrij** [comment]
@donadigo done

**Comment by donadigo** [comment]
@artemanufrij you didn't remove the files themselves.

**Comment by donadigo** [comment]
@artemanufrij just do:
`git rm .elementary-ide.proj`
`git rm src/config.vala`

and then commit.

**Comment by artemanufrij** [comment]
@donadigo sorry... done

**Pull request merged**